synapse/lib/node.py

```diff
@@ -777,6 +777,12 @@ async def delete(self, force=False):
         await self.snap.applyNodeEdit((self.buid, formname, edits))
         self.snap.livenodes.pop(self.buid, None)
 
+    async def hasData(self, name):
+        if name in self.nodedata:
+            return True
+        node = await self.snap.getNodeByBuid(self.buid)
```

Comment:
We may need to plumb a hasNodeData call down into the snap since lifting a node doesn't also lift node data automatically.  The tests below are working because you set the nodedata on that exact node, but would likely fail if you ran the has() in a subsequent query.

Reply:
self.nodedata is a cache of whatever we happen to have already lifted or set but not a copy of everything.

Reply:
Ah. I saw the potion in the snap._joinSodes about merging together the nodedata and thought the lift automatically grabbed that too. And you're right. It doesn't work on subsequent query. But shouldn't be too awful to add.